dedalus/core/arithmetic.py

```diff
@@ -467,7 +467,7 @@ def expression_matrices(self, subproblem, vars, **kw):
 
     def matrix_dependence(self, *vars):
         coupling = self.matrix_coupling(*vars)
-        coupling[1] = True  # HACK HACK HACK for spheres coupling ell
+        #coupling[1] = True  # HACK HACK HACK for spheres coupling ell
```

Comment:
Uncommenting this line causes `test_ball_ncc.py` to work again and pass all tests, but this line does not work with `test_nlbvp.py`.